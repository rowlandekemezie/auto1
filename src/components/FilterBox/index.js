import React from "react";
import Select from "react-select";
import styled from "styled-components";
import PropTypes from "prop-types";

import colors from "../../static/colors";
import { capitalize } from "../../utils";

const Wrapper = styled("div")``;

const Text = styled("p")``;

export default function FilterBox({
  handleOnChange,
  value,
  title,
  options = []
}) {
  return (
    <Wrapper>
      <Text>{capitalize(title)}</Text>
      <Select
        value={value}
        options={options}
        onChange={handleOnChange}
        theme={theme => ({
          ...theme,
          colors: {
            ...theme.colors,
            primary25: colors.LIGHT_ORANGE,
            primary: colors.LIGHT_ORANGE
          }
        })}
      />
    </Wrapper>
  );
}

FilterBox.propTypes = {
  value: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired
  }),
  handleOnChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ),
  title: PropTypes.string
};
