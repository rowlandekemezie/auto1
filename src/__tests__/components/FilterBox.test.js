import React from "react";
import { shallow, mount } from "enzyme";
import FilterBox from "../../components/FilterBox";

describe("<FilterBox />", () => {
  const props = {
    title: "Test",
    value: { label: "test", value: "" },
    handleOnChange: jest.fn()
  };
  it("should render component properly", () => {
    const wrapper = shallow(<FilterBox {...props} />);
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it("should props with the right values", () => {
    const wrapper = mount(<FilterBox {...props} />);
    expect(wrapper.find("p").length).toBe(1);
    expect(wrapper.find("p").text()).toBe(props.title);
  });
});
