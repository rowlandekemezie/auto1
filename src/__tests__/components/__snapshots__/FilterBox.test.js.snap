// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterBox /> should render component properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBox
    handleOnChange={[MockFunction]}
    title="Test"
    value={
      Object {
        "label": "test",
        "value": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Test
        </ForwardRef>,
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[MockFunction]}
          options={Array []}
          theme={[Function]}
          value={
            Object {
              "label": "test",
              "value": "",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [],
          },
          "displayName": "styled.p",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultInputValue": "",
          "defaultMenuIsOpen": false,
          "defaultValue": null,
          "onChange": [MockFunction],
          "options": Array [],
          "theme": [Function],
          "value": Object {
            "label": "test",
            "value": "",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Test
          </ForwardRef>,
          <StateManager
            defaultInputValue=""
            defaultMenuIsOpen={false}
            defaultValue={null}
            onChange={[MockFunction]}
            options={Array []}
            theme={[Function]}
            value={
              Object {
                "label": "test",
                "value": "",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [],
            },
            "displayName": "styled.p",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "p",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultInputValue": "",
            "defaultMenuIsOpen": false,
            "defaultValue": null,
            "onChange": [MockFunction],
            "options": Array [],
            "theme": [Function],
            "value": Object {
              "label": "test",
              "value": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
